spring-xd-dirt/src/main/java/org/springframework/xd/dirt/job/dsl/Graph.java

```diff
@@ -200,23 +210,43 @@ else if (toFollow.size() == 1) { // FLOW
 				Link linkToFollow = toFollow.get(0);
 				Node linkToFollowTarget = findNodeById(linkToFollow.to);
 				if (linkToFollowTarget != nodeToTerminateFollow) {
-					if (linkToFollowTarget.isEnd() || linkToFollowTarget.isFail()) {
-						unfollowedLinks.remove(linkToFollow);
-					}
-					else {
-						if (graphText.length() != 0) {
-							// First one doesn't need a || on the front
-							graphText.append(" || ");
-						}
-						followLink(graphText, linkToFollow, nodeToTerminateFollow, unvisitedNodes, unfollowedLinks);
+					// need special handling for end/fail??
+					if (graphText.length() != 0) {
+						// First one doesn't need a || on the front
+						graphText.append(" || ");
 					}
+					followLink(graphText, linkToFollow, nodeToTerminateFollow, unvisitedNodes, unfollowedLinks);
 				}
 				break;
 			}
 		}
 	}
 
+	private String toString(Link link) {
+		StringBuilder s = new StringBuilder();
```

Comment:
Not sure we need to use StringBuilder in this method to create our string in this method.